src/main/resources/xslt/preservica2schemaorg.xsl

```diff
@@ -40,69 +42,168 @@
         </xsl:choose>
       </xsl:variable>
 
-      <f:map>
-        <f:string key="@context">http://schema.org/</f:string>
-        <f:string key="@type"><xsl:value-of select="$type"/></f:string>
-        <f:string key="id">
-            <xsl:value-of select="$recordID"/>
-        </f:string>
-
-        <!-- Extract PBCore metadata -->
-        <xsl:for-each select="/xip:DeliverableUnit/Metadata/pbc:PBCoreDescriptionDocument">
-          <xsl:call-template name="pbc-metadata">
+      <!-- Choose which type of transformation to do, based on the input data.
+           This choose-statement decides, which of the following templates are to be used for the given record. -->
+      <xsl:choose>
+        <xsl:when test="$type = 'VideoObject'">
+          <xsl:call-template name="video-transformation">
             <xsl:with-param name="type" select="$type"/>
             <xsl:with-param name="pbcExtensions" select="$pbcExtensions"/>
           </xsl:call-template>
-        </xsl:for-each>
+        </xsl:when>
+        <xsl:when test="$type = 'AudioObject'">
+          <xsl:call-template name="audio-transformation">
+            <xsl:with-param name="type" select="$type"/>
+            <xsl:with-param name="pbcExtensions" select="$pbcExtensions"/>
+          </xsl:call-template>
+        </xsl:when>
+        <xsl:otherwise>
+          <xsl:call-template name="generic-transformation">
+            <xsl:with-param name="type" select="$type"/>
+            <xsl:with-param name="pbcExtensions" select="$pbcExtensions"/>
+          </xsl:call-template>
+        </xsl:otherwise>
+      </xsl:choose>
```

Comment:
It looks like both `VideoObject` and `AudioObject` calls the same templates as `generic-transformation` + some extra? Couldn't `generic-transformation` be called for everything and then the video/audio-specific parts could be called?

Reply:
AudioObject can actually call it directly! As I see it, the video Object cannot make use of it as some of the values are represented in a map that gets closed in to generic-template. I could leave this open to make it possible to add to it. However, that would require, that all templates remembers to close that map after calling the generic-template